### PR TITLE
Add ternary operators

### LINKED ISSUE #41: Ternary Expression
@alexschneider

### PR DESCRIPTION
**Pull request by alexschneider**
Also with bonus fallthrough!

I cleaned up one or two ASTs as well - not really worth a whole new PR. 

Fixes #41

### CODE REVIEW

File: test/example/input_programs/valid_programs/program15.tea [hunk 1]
```diff
@@ -0,0 +1 @@
+a if b else c
```
Comment (rachelriv) [inline test/example/input_programs/valid_programs/program15.tea:1]: We should try to think of some more edge cases later on if possible.
